### PR TITLE
recyclerview adapter rendring

### PR DESCRIPTION
**Pull request by kuldeep-mobi**

### CODE REVIEW

File: app/src/main/res/layout/item_food.xml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+<layout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    xmlns:tools="http://schemas.android.com/tools">
+    <data >
+        <variable
+            name="mRecipes"
+            type="com.mvvmdemo.model.Recipes"/>
+    </data>
+
+    <androidx.cardview.widget.CardView
```
Comment (rish22abh) [inline app/src/main/res/layout/item_food.xml:13]: Use LinearLayout. Coz Relative layout more time to render.

File: app/src/main/java/com/mvvmdemo/activity/MainActivity.kt [hunk 2]
```diff
@@ -32,6 +33,7 @@ class MainActivity : BaseActivity() {
 
         mainViewModel.getSearchApi().observe(this, Observer {
             Log.e("Rishabh", "obser-$it")
+            binding.mAdapter = FoodAdapter(this,it!!.recipes)
```
Comment (rish22abh) [inline app/src/main/java/com/mvvmdemo/activity/MainActivity.kt]: Don't set adapter.. here only notify adapter.. Initialize adapter outside observer.

File: app/src/main/java/com/mvvmdemo/adapters/BindingAdapter.kt [hunk 3]
```diff
@@ -0,0 +1,10 @@
+package com.mvvmdemo.adapters
+
+import android.widget.ImageView
+import androidx.databinding.BindingAdapter
+import com.bumptech.glide.Glide
+
+@BindingAdapter("loadimage")
+fun loadImage(mView : ImageView, str:String){
+    Glide.with(mView.context).load(str).into(mView)
```
Comment (rish22abh) [inline app/src/main/java/com/mvvmdemo/adapters/BindingAdapter.kt]: Create Handler for Glide.

File: .idea/.name [hunk 4]
```diff
@@ -0,0 +1 @@
+MvvmDemo
```
Comment (rish22abh) [inline .idea/.name:1]: delete this file

File: app/build.gradle [hunk 5]
```diff
@@ -40,4 +40,7 @@ dependencies {
     testImplementation 'junit:junit:4.13'
     androidTestImplementation 'androidx.test.ext:junit:1.1.1'
     androidTestImplementation 'androidx.test.espresso:espresso-core:3.2.0'
+    implementation 'com.github.bumptech.glide:glide:4.11.0'
+    annotationProcessor 'com.github.bumptech.glide:compiler:4.11.0'
```
Comment (rish22abh) [inline app/build.gradle]: insist of annotationProcessor use kapt

File: app/src/main/java/com/mvvmdemo/adapters/FoodAdapter.kt [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+    override fun getItemCount(): Int {
+       return mList.size
+    }
+
+    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
+        holder.bind(mList[position])
+    }
+
+    class ViewHolder(private val itemFoodBinding: ItemFoodBinding):RecyclerView.ViewHolder(itemFoodBinding.root) {
+        fun bind(mRecipes : Recipes){
```
Comment (rish22abh) [inline app/src/main/java/com/mvvmdemo/adapters/FoodAdapter.kt]: no need of bind method. use directly in onBindViewHolder

File: app/src/main/java/com/mvvmdemo/activity/MainActivity.kt (hunk 2 above)
Comment (kuldeep-mobi) [reply app/src/main/java/com/mvvmdemo/activity/MainActivity.kt]: Done

File: app/src/main/java/com/mvvmdemo/adapters/FoodAdapter.kt (hunk 6 above)
Comment (kuldeep-mobi) [reply app/src/main/java/com/mvvmdemo/adapters/FoodAdapter.kt]: No Need.

File: app/build.gradle (hunk 5 above)
Comment (kuldeep-mobi) [reply app/build.gradle]: Done.

**Review by rish22abh — APPROVED** [review]

**Pull request merged**